app/uk/gov/hmrc/openidconnect/userinfo/services/UserInfoTransformer.scala

```diff
@@ -51,19 +51,19 @@ trait UserInfoTransformer {
     val identifier = if (scopes.contains("openid:gov-uk-identifiers")) Some(nino) else None
     val address = if (scopes.contains("address")) desUserInfo map (u => Address(formattedAddress(u.address, country), u.address.postcode, country)) else None
 
-    UserInfo(profile.map(_.firstName),
-      profile.map(_.familyName),
+    UserInfo(profile.flatMap(_.firstName),
+      profile.flatMap(_.familyName),
       profile.flatMap(_.middleName),
       address,
       profile.flatMap(_.birthDate),
       identifier)
   }
 
   private def formattedAddress(desAddress: DesAddress, country: Option[String]) = {
-    s"${desAddress.line1}\n${desAddress.line2}${addLine(desAddress.line3)}${addLine(desAddress.line4)}${addLine(desAddress.postcode)}${addLine(country)}"
+    s"${desAddress.line1.get}${addLine(desAddress.line2)}${addLine(desAddress.line3)}${addLine(desAddress.line4)}${addLine(desAddress.postcode)}${addLine(country)}"
```

Comment:
This isn't safe - we'll still get errors thrown when creating a formatted address when there is no line1 or line2 addresses.

It would be better to do: 

```
 +    s"${addLine(desAddress.line1}${addLine(desAddress.line2)}${addLine(desAddress.line3)}${addLine(desAddress.line4)}${addLine(desAddress.postcode)}${addLine(country)}" 
```

Perhaps some 'parsing addresses' unit tests would help with catching this sort of thing?